test/graphql_test.rb

```diff
@@ -374,4 +374,35 @@ def test_mutation_batch_subselection_execution
     assert_equal expected, result
     assert_equal ["Product/1,2", "Product/2,3"], queries
   end
+
+  class MyCustomExecutor < GraphQL::Batch::Executor
+    attr_reader :loaders_executed
+
+    def initialize
+      @loaders_executed = []
+      super
+    end
+
+    def around_promise_callbacks(loader)
+      @loaders_executed << loader
+      super
+    end
+  end
+
+  def test_custom_executor_class
+    schema = GraphQL::Schema.define do
+      query ::QueryType
+      mutation ::MutationType
+
+      use GraphQL::Batch, executor_class: MyCustomExecutor
+    end
+
+    custom_executor = MyCustomExecutor.new
+    MyCustomExecutor.stubs(:new).returns(custom_executor)
+
+    query_string = '{ product(id: "1") { id } }'
+    schema.execute(query_string)
+
+    refute custom_executor.loaders_executed.empty?
```

Comment:
Possibly related to https://github.com/Shopify/shopify/pull/136940#issuecomment-342542248?
I noticed that this ended up containing two references to the same instance of RecordLoader.